lib/porky_lib/symmetric.rb

```diff
@@ -191,6 +191,40 @@ def secure_delete_plaintext_key(length)
     "\0" * length
   end
 
+  def encrypt_with_key(plaintext, plaintext_key)
+    # Initialize the box
+    secret_box = RbNaCl::SecretBox.new(plaintext_key)
```

Comment:
isn't this part very similar to what we have in encrypt? Maybe modify encrypt to use this method?

Reply:
It's the same, yes. I'm reluctant to modify the existing functions in case of unforeseen complications. 

If we feel confident in make this change, however, I can do so.

Reply:
I would be too but it feels duplicated, fine with it